src/System.Collections.NonGeneric/tests/Hashtable/PropertyItemThreadSafety.cs

```diff
@@ -41,46 +42,47 @@ public void TestGetItemThreadSafety()
 
         //Setup key1 and key2 so they are different values but have the same hashcode
         //To produce a hashcode long XOR's the first 32bits with the last 32 bits
-        l1 = (((long)i1) << 32) + i2;
-        l2 = (((long)i2) << 32) + i1;
-        _key1 = (object)l1;
-        _key2 = (object)l2;
+        l1 = (((long) i1) << 32) + i2;
+        l2 = (((long) i2) << 32) + i1;
+        _key1 = (object) l1;
+        _key2 = (object) l2;
 
         _ht = new Hashtable(3); //Just one item will be in the hashtable at a time
         int taskCount = 3;
         Task[] readers1 = new Task[taskCount];
         Task[] readers2 = new Task[taskCount];
-            Task writer;
-            DateTime startTime = DateTime.Now;
+        Task writer;
 
-            for (int i = 0; i < readers1.Length; i++)
-            {
-                readers1[i] = Task.Run(new Action(ReaderFunction1));
-            }
+        Stopwatch stopwatch = Stopwatch.StartNew();
 
-            for (int i = 0; i < readers2.Length; i++)
-            {
-                readers2[i] = Task.Run(new Action(ReaderFunction2));
-            }
+        for (int i = 0; i < readers1.Length; i++)
+        {
+            readers1[i] = Task.Run(new Action(ReaderFunction1));
+        }
+
+        for (int i = 0; i < readers2.Length; i++)
+        {
+            readers2[i] = Task.Run(new Action(ReaderFunction2));
+        }
 
-            writer = Task.Run(new Action(WriterFunction));
+        writer = Task.Run(new Action(WriterFunction));
 
-            SpinWait spin = new SpinWait();
-            while (!_errorOccured && !_timeExpired)
+        SpinWait spin = new SpinWait();
+        while (!_errorOccured && !_timeExpired)
+        {
+            if (MAX_TEST_TIME_MS < stopwatch.ElapsedMilliseconds)
             {
-                if (MAX_TEST_TIME_TICKS < DateTime.Now.Ticks - startTime.Ticks)
-                {
-                    _timeExpired = true;
-                }
-
-                spin.SpinOnce();
+                _timeExpired = true;
             }
 
-            Task.WaitAll(readers1);
-            Task.WaitAll(readers2);
-            writer.Wait();
+            spin.SpinOnce();
```

Comment:
Separate from your fix, we should rewrite this test (and any others like it)... there are better ways to construct this that don't involve spinning for 10 seconds.